pysd/py_backend/output.py

```diff
@@ -0,0 +1,519 @@
+        self.ds.final_time = f"{model.components.final_time()}"
+
+        # creating variables for all model dimensions
+        for dim_name, coords in model.subscripts.items():
+            coords = np.array(coords)
+            # create dimension
+            self.ds.createDimension(dim_name, len(coords))
+            # length of the longest string in the coords
+            max_str_len = len(max(coords, key=len))
+
+            # create variable
+            # TODO: check if the type could be defined otherwise)
+            var = self.ds.createVariable(
+                dim_name, f"S{max_str_len}", (dim_name,))
+            # assigning values to variable
+            var[:] = coords
+
+        # creating the time dimension as unlimited
+        self.ds.createDimension("time", None)
+
+        # creating variables in capture_elements
+        self.__create_ds_vars(model, capture_elements)
+
+    def update(self, model, capture_elements):
+        """
+        Writes values of variables in capture_elements in the netCDF4 Dataset.
+
+        Parameters
+        ----------
+        model: pysd.Model
+            PySD Model object
+        capture_elements: set
+            Which model elements to capture - uses pysafe names.
+
+        Returns
+        -------
+        None
+        """
+        for key in capture_elements:
+
+            comp = getattr(model.components, key)
+            comp_vals = comp()
+
+            if "time" in self.ds[key].dimensions:
+                if isinstance(comp_vals, xr.DataArray):
+                    self.ds[key][self.step, :] = comp_vals.values
+                elif isinstance(comp_vals, np.ndarray):
+                    self.ds[key][self.step, :] = comp_vals
+                else:
+                    self.ds[key][self.step] = comp_vals
+            else:
+                try:  # this issue can arise with external objects
+                    if isinstance(comp_vals, xr.DataArray):
+                        self.ds[key][:] = comp_vals.values
+                    elif isinstance(comp_vals, np.ndarray):
+                        if comp_vals.size == 1:
+                            self.ds[key][:] = comp_vals
+                        else:
+                            self.ds[key][:] = comp_vals
+                    else:
+                        self.ds[key][:] = comp_vals
+                except ValueError:
+                    warnings.warn(
+                        f"The dimensions of {key} in the results "
+                        "do not match the declared dimensions for this "
+                        "variable. The resulting values will not be "
+                        "included in the results file.")
+
+        self.step += 1
+
+    def postprocess(self, **kwargs):
+        """
+        Closes netCDF4 Dataset.
+
+        Returns
+        -------
+        None
+        """
+
+        # close Dataset
+        self.ds.close()
+
+        if kwargs.get("flatten"):
```

Comment:
flatten is by default True, Having this warning always will be tiring.
I would just mention in the `.run` method documentation that flatten is only used for pd.DataFrame output. Otherwise, just ignore it, this approach is commonly used by several libraries.

Reply:
done